tests/FakeItEasy.Analyzer.VisualBasic.Tests/ArgumentConstraintNullabilityMismatchAnalyzerTests.cs

```diff
@@ -0,0 +1,231 @@
+        {
+            string completeConstraint = "A(Of Integer).Ignored";
+            string call = $"foo({completeConstraint})";
+            string code = string.Format(CodeTemplate, call);
+
+            string fixedConstraint = "A(Of Integer?).Ignored";
+            string fixedCall = $"foo({fixedConstraint})";
+            string fixedCode = string.Format(CodeTemplate, fixedCall);
+            this.VerifyVisualBasicFix(code, fixedCode, codeFixIndex: 0);
+        }
+
+        [Fact]
+        public void MakeConstraintNullable_CodeFix_should_replace_constraint_with_nullable_constraint_for_named_indexer()
+        {
+            string completeConstraint = "A(Of Integer).Ignored";
+            string call = $"foo.Item({completeConstraint})";
+            string code = string.Format(CodeTemplate, call);
+
+            string fixedConstraint = "A(Of Integer?).Ignored";
+            string fixedCall = $"foo.Item({fixedConstraint})";
+            string fixedCode = string.Format(CodeTemplate, fixedCall);
+            this.VerifyVisualBasicFix(code, fixedCode, codeFixIndex: 0);
+        }
+
+        [Fact]
+        public void MakeNotNullConstraint_CodeFix_should_replace_constraint_with_AThatIsNotNull()
+        {
+            string completeConstraint = "A(Of Integer).Ignored";
+            string call = $"foo.NullableParam({completeConstraint})";
+            string code = string.Format(CodeTemplate, call);
+
+            string fixedConstraint = "A(Of Integer?).That.IsNotNull()";
+            string fixedCall = $"foo.NullableParam({fixedConstraint})";
+            string fixedCode = string.Format(CodeTemplate, fixedCall);
+            this.VerifyVisualBasicFix(code, fixedCode, codeFixIndex: 1);
+        }
+
+        [Fact]
+        public void MakeNotNullConstraint_CodeFix_should_replace_constraint_with_AThatIsNotNull_for_indexer()
+        {
+            string completeConstraint = "A(Of Integer).Ignored";
+            string call = $"foo({completeConstraint})";
+            string code = string.Format(CodeTemplate, call);
+
+            string fixedConstraint = "A(Of Integer?).That.IsNotNull()";
+            string fixedCall = $"foo({fixedConstraint})";
+            string fixedCode = string.Format(CodeTemplate, fixedCall);
+            this.VerifyVisualBasicFix(code, fixedCode, codeFixIndex: 1);
+        }
+
+        [Fact]
+        public void MakeNotNullConstraint_CodeFix_should_replace_constraint_with_AThatIsNotNull_for_named_indexer()
+        {
+            string completeConstraint = "A(Of Integer).Ignored";
+            string call = $"foo.Item({completeConstraint})";
+            string code = string.Format(CodeTemplate, call);
+
+            string fixedConstraint = "A(Of Integer?).That.IsNotNull()";
+            string fixedCall = $"foo.Item({fixedConstraint})";
+            string fixedCode = string.Format(CodeTemplate, fixedCall);
+            this.VerifyVisualBasicFix(code, fixedCode, codeFixIndex: 1);
+        }

```

Comment:
There are more tests in VB.NET because there are two way to invoke indexers: using the default property (`foo(index)`) or using a named indexer (`foo.Item(index)`).